### PR TITLE
libmapi: Add get_TimeZoneDefinition function

### PR DESCRIPTION
**Pull request by icandigitbaby**
There was a setter for the TimeZoneDefinition structure, but not a
getter. This method converts a binary blob into the structure and can be
used to interpret properties such as
PidLidAppointmentTimeZoneDefinitionRecur,
PidLidAppointmentTimeZoneDefinitionStartDisplay and
PidLidAppointmentTimeZoneDefinitionEndDisplay.

In addition, the biases in the time zone structures were declared as unsigned integers when they can be negative. I haven't seen this causing any bug, but I made a change to make the declaration correct.

### CODE REVIEW

**Comment by icandigitbaby** [comment]
@sixstone-qq The specification doesn't say anywhere that this bias can be negative, nor which type it has (it only says that it's a field of 2 bytes), biases just happen to be negative in western time zones, and that's how the client sends them. I added the sections where the structures are specified, though.

File: libmapi/property.c [hunk 1]
```diff
@@ -1614,6 +1614,49 @@ _PUBLIC_ struct Binary_r *set_TimeZoneStruct(TALLOC_CTX *mem_ctx, const struct T
...
+	struct TimeZoneDefinition	*TimeZoneDefinition = NULL;
+	struct ndr_pull			*ndr;
+	enum ndr_err_code		ndr_err_code;
+
+	/* Sanity checks */
+	if (!bin) return NULL;
+	if (!bin->cb) return NULL;
+	if (!bin->lpb) return NULL;
+
+	ndr = talloc_zero(mem_ctx, struct ndr_pull);
```
Comment (sixstone-qq) [inline libmapi/property.c]: Missing OOM check

File: libmapi/property.c [hunk 2]
```diff
@@ -1614,6 +1614,49 @@ _PUBLIC_ struct Binary_r *set_TimeZoneStruct(TALLOC_CTX *mem_ctx, const struct T
...
+	ndr->offset = 0;
+	ndr->data = bin->lpb;
+	ndr->data_size = bin->cb;
+
+	ndr_set_flags(&ndr->flags, LIBNDR_FLAG_NOALIGN);
+	TimeZoneDefinition = talloc_zero(mem_ctx, struct TimeZoneDefinition);
```
Comment (sixstone-qq) [inline libmapi/property.c]: Missing OOM check

**Comment by sixstone-qq** [comment]
@icandigitbaby it is specified in the examples [MS-OXOCAL] Sections 4.1.4 and 4.1.5 where it says they are LONG (MS-DTYP) to which we have to interpret as signed 32-bit integer.

Apart from the minor comments, :+1:

**Comment by icandigitbaby** [comment]
Cheers!

**Pull request merged**